oauth2_http/java/com/google/auth/oauth2/MetricsUtils.java

```diff
@@ -67,4 +71,50 @@ private static String getAuthLibraryVersion() {
     }
     return version;
   }
+
+  public enum RequestType {
+    ACCESS_TOKEN_REQUEST("at"),
+    ID_TOKEN_REQUEST("it"),
+    METADATA_SERVER_PIN("mds"),
+    UNSPECIFIED("unspecified"), // should not send request type info
+    UNTRACKED("untracked"); // should not add metric header
```

Comment:
nit: If the change above for ComputeEngineCredentials makes sense, UNTRACKED sounds more accurate to me. The RequestType isn't tracked and therefore isn't added. Would it be possible to swap UNSPECIFIED and UNTRACKED?